mantidproject/mantid
# Use property manager in batch algorithm runner

Note that this work is by @DavidFair and @ConorMFinn rather than myself.

This PR makes changes to the BatchAlgorithmRunner and related classes to change from using AlgorithmRuntimeProps which was previously a map of string to string, to using the property manager so that we can set proper property types. This is required for upstream work in the reflectometry GUI which will need to pass workspaces as pointers rather than string names, because they will not be in the ADS.

**To test:**

Check that:
- Test something in Reflectometry GUI that relies on it still works
  - Open the ISIS Reflectometry GUI
  - Set instrument INTER. Enter run number 13460, angle 0.4
  - Click process. The row should turn green when processing completes successfully.
- Test something in Indirect (from their testing instructions) still works

Refs #32772

*This does not require release notes* because **it is an internal changes that does not affect functionality**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
